api/src/client/DynamicsAccessTokenClient.test.ts

```diff
@@ -0,0 +1,82 @@
+import axios from "axios";
+import { AccessTokenClient } from "../domain/types";
+import { LogWriter, LogWriterType } from "../libs/logWriter";
+import { DynamicsAccessTokenClient } from "./DynamicsAccessTokenClient";
+
+jest.mock("winston");
+jest.mock("axios");
+const mockAxios = axios as jest.Mocked<typeof axios>;
+
+describe("DynamicsAccessTokenClient", () => {
+  let client: AccessTokenClient;
+  let logger: LogWriterType;
+
+  let ORIGINAL_ORG_URL: string | undefined;
+  const ORG_URL = "www.test-org-url.com";
+
+  let ORIGINAL_TENANT_ID: string | undefined;
+  const TENANT_ID = "test-tenant-id-123";
+
+  let ORIGINAL_CLIENT_ID: string | undefined;
+  const CLIENT_ID = "test-client-id-123";
+
+  let ORIGINAL_CLIENT_SECRET: string | undefined;
+  const CLIENT_SECRET = "test-client-secret-123";
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    logger = LogWriter("NavigatorWebService", "ApiLogs", "us-test-1");
+
+    ORIGINAL_ORG_URL = process.env.DCA_DYNAMICS_ORG_URL;
+    process.env.DCA_DYNAMICS_ORG_URL = ORG_URL;
+
+    ORIGINAL_TENANT_ID = process.env.DCA_DYNAMICS_TENANT_ID;
+    process.env.DCA_DYNAMICS_TENANT_ID = TENANT_ID;
+
+    ORIGINAL_CLIENT_ID = process.env.DCA_DYNAMICS_CLIENT_ID;
+    process.env.DCA_DYNAMICS_CLIENT_ID = CLIENT_ID;
+
+    ORIGINAL_CLIENT_SECRET = process.env.DCA_DYNAMICS_SECRET;
+    process.env.DCA_DYNAMICS_SECRET = CLIENT_SECRET;
+
+    client = DynamicsAccessTokenClient(logger, {
+      tenantId: process.env.DCA_DYNAMICS_TENANT_ID || "",
+      orgUrl: process.env.DCA_DYNAMICS_ORG_URL || "",
+      clientId: process.env.DCA_DYNAMICS_CLIENT_ID || "",
+      clientSecret: process.env.DCA_DYNAMICS_SECRET || "",
+    });
+  });
+
+  afterEach(() => {
+    process.env.DCA_DYNAMICS_ORG_URL = ORIGINAL_ORG_URL;
+    process.env.DCA_DYNAMICS_TENANT_ID = ORIGINAL_TENANT_ID;
+    process.env.DCA_DYNAMICS_CLIENT_ID = ORIGINAL_CLIENT_ID;
+    process.env.DCA_DYNAMICS_SECRET = ORIGINAL_CLIENT_SECRET;
```

Comment:
[pebble] I think this whole process of setting the env vars in `process.env` is unnecessary, because the code under test isn't pulling them out of the environment, they're getting passed in as config vars to the Client. And that's good! It means that the client doesn't depend on the node environment. But the tests here can be significantly simpler because you just need to pass the string into the client, you shouldn't need anything to do anything with `process.env` at all.

Reply:
Same comment in the other tests also.

Reply:
I originally wasn't passing them in so this made sense but I should have changed it to your point after the refactor to pass them in. Good call out! I changed them all expect for `searchLicenseStatusFactory` because that actually depends on the feature flag to determine whether to use Dynamics or not, so I left that one the same.